lib/web/scripts.go

```diff
@@ -0,0 +1,149 @@
+// or configure its services. This is handled by the "/scripts/:token/install-*.sh" scripts.
+func (h *Handler) installScriptHandle(w http.ResponseWriter, r *http.Request, params httprouter.Params) (any, error) {
+	// TODO(hugoShaka): cache function
+	opts, err := h.installScriptOptions(r.Context())
+	if err != nil {
+		return nil, trace.Wrap(err, "Failed to build install script options")
+	}
+
+	script, err := scripts.GetInstallScript(r.Context(), opts)
+	if err != nil {
+		h.logger.WarnContext(r.Context(), "Failed to get install script", "error", err)
+		return nil, trace.Wrap(err, "getting script")
+	}
+
+	w.WriteHeader(http.StatusOK)
+	if _, err := fmt.Fprintln(w, script); err != nil {
+		h.logger.WarnContext(r.Context(), "Failed to write install script", "error", err)
+	}
+
+	return nil, nil
+}
+
+// installScriptOptions computes the agent installation options based on the proxy configuration and the cluster status.
+// This includes:
+// - the type of automatic updates
+// - the desired version
+// - the proxy address (used for updates).
+// - the Teleport artifact name and CDN
+func (h *Handler) installScriptOptions(ctx context.Context) (scripts.InstallScriptOptions, error) {
+	const group, uuid = "", ""
+
+	version, err := h.autoUpdateAgentVersion(ctx, group, uuid)
+	if err != nil {
+		h.logger.WarnContext(ctx, "Failed to update agent version", "error", err)
+		version = teleport.Version
+	}
+
+	// if there's a rollout, we do new autoupdates
+	_, rolloutErr := h.cfg.AccessPoint.GetAutoUpdateAgentRollout(ctx)
+	if rolloutErr != nil && !trace.IsNotFound(rolloutErr) {
+		h.logger.WarnContext(ctx, "Failed to get rollout", "error", rolloutErr)
+		return scripts.InstallScriptOptions{}, trace.Wrap(err, "failed to check the autoupdate agent rollout state")
+	}
+
+	var autoupdateStyle scripts.AutoupdateStyle
+	switch {
+	case rolloutErr == nil:
+		autoupdateStyle = scripts.UpdaterBinaryAutoupdate
+	case automaticUpgrades(h.clusterFeatures):
+		autoupdateStyle = scripts.PackageManagerAutoupdate
+	default:
+		autoupdateStyle = scripts.NoAutoupdate
+	}
+
+	var teleportFlavor string
+	switch modules.GetModules().BuildType() {
+	case modules.BuildEnterprise:
+		teleportFlavor = types.PackageNameEnt
+	case modules.BuildOSS, modules.BuildCommunity:
+		teleportFlavor = types.PackageNameOSS
+	default:
+		h.logger.WarnContext(ctx, "Unknown built type, defaulting to the 'teleport' package.", "type", modules.GetModules().BuildType())
+		teleportFlavor = types.PackageNameOSS
+	}
+
+	cdnBaseURL, err := getCDNBaseURL()
+	if err != nil {
+		h.logger.WarnContext(ctx, "Failed to get CDN base URL", "error", err)
+		return scripts.InstallScriptOptions{}, trace.Wrap(err)
+	}
+
+	return scripts.InstallScriptOptions{
+		AutoupdateStyle: autoupdateStyle,
+		TeleportVersion: version,
+		// Note: this override is required to configure the license on AGPL builds.
+		// We cannot install random binaries if the user doesn't
```

Comment:
Which override? the CDN Base URL?
Also, feels like the comment was not completed.